yang/goldstone-static-route.yang

```diff
@@ -0,0 +1,127 @@
+module goldstone-static-route {
+
+  yang-version "1";
+
+  namespace "http://goldstone.net/yang/goldstone-static-route";
+  prefix gs-static-route;
+
+  import goldstone-routing {
+    prefix gs-rt;
+  }
+
+  import goldstone-nexthop {
+    prefix gs-nexthop;
+  }
+
+  import ietf-inet-types {
+    prefix inet;
+  }
+
+  organization
+    "GoldStone";
+
+  description
+    "This module contains a collection of YANG definitions for
+    managing and configuring static route";
+
+  revision 2022-12-14 {
+    description
+      "Initial revision.";
+  }
+
+  grouping path-list-config {
+    description
+      "path-list configuration";
+
+    leaf table-id {
+      type uint32;
+      description
+        "Table-id";
+    }
+
+    leaf distance {
+      type gs-rt:administrative-distance;
+      description
+        "Admin distance associated with this route.";
+    }
+
+    leaf tag {
+      type uint32;
+      default "0";
+      description
+        "Route tag";
+    }
+  }
+
+  grouping static-route-prefix-attributes {
+    list path-list {
+      key "table-id distance";
```

Comment:
I do not know the advantage of having a path list indexed by `table-id` and `distance`, but it sounds strange to me. I have not seen also this kind of structure in other vendors' routing modules.
Even in FRR, table-id and distance are not mandatory for static-route configuration.
I would like to suggest removing the path-list and moving the attributes `distance` to the next-hop entry configuration and `tag` to the route configuration context.

Reply:
@ipi-claytonpascoal 
I tried to consider the model based on your comments.
Removed the `path-list` and moved `distance` to next-hop entry.
`Tag` may be used when we support OSPFv2, so it is ignored now.
Do you agree with this structure?

```
module: goldstone-static-route

  augment /gs-rt:routing/gs-rt:control-plane-protocol:
    +--rw static-route
       +--rw route-list* [prefix]
          +--rw prefix      -> ../config/prefix
          +--rw config
          |  +--rw prefix?   inet:ipv4-prefix
          +--ro state
          |  +--ro prefix?   inet:ipv4-prefix
          +--rw nexthops
             +--rw nexthop* [nexthop-type gateway interface]
                +--rw nexthop-type    -> ../config/nexthop-type
                +--rw gateway         -> ../config/gateway
                +--rw interface       -> ../config/interface
                +--rw config
                |  +--rw nexthop-type      nexthop-type
                |  +--rw gateway?          optional-ip-address
                |  +--rw interface?        gs-if:interface-ref
                |  +--rw distance?         gs-rt:administrative-distance
                |  +--rw blackhole-type?   blackhole-type
                |  +--rw onlink?           boolean
                +--ro state
                   +--ro nexthop-type      nexthop-type
                   +--ro gateway?          optional-ip-address
                   +--ro interface?        gs-if:interface-ref
                   +--ro distance?         gs-rt:administrative-distance
                   +--ro blackhole-type?   blackhole-type
                   +--ro onlink?           boolean
                   +--ro active?           boolean
                   +--ro fib?              boolean
```

Reply:
It sounds pretty good!
I just added my considerations regarding nexthop list keys [above](https://github.com/oopt-goldstone/goldstone-mgmt/pull/73/files#r1054965093)